Modules/Core/Sources/Core/Stream/Stream.swift

```diff
@@ -1,28 +1,44 @@
 public enum StreamError : Error {
-    case closedStream(buffer: Buffer)
-    case timeout(buffer: Buffer)
+    case closedStream
+    case timeout
 }
 
 public protocol InputStream {
     var closed: Bool { get }
+    func open(deadline: Double) throws
     func close()
     
-    func read(into: UnsafeMutableBufferPointer<UInt8>, deadline: Double) throws -> Int
-    func read(upTo: Int, deadline: Double) throws -> Buffer
+    func read(into readBuffer: UnsafeMutableBufferPointer<Byte>, deadline: Double) throws -> UnsafeBufferPointer<Byte>
```

Comment:
Interesting implementation that returns a `UnsafeBufferPointer` rather than a count. I actually like this as it is necessary on nearly all call sites